resources/gui/js/apis/utils.js

```diff
@@ -85,21 +85,67 @@ class UtilsApi extends IApi {
    * @return {string}
    */
   formatNumber(number) {
-    number = Number(number);
+    const abs = Math.abs(number);
 
-    // Set very small numbers to 0
-    if (number < Number.EPSILON && number > -Number.EPSILON) {
-      number = 0;
+    if (abs >= 10000) {
+      return number.toPrecision(2);
+    }
+    if (abs >= 100) {
+      return Number(number.toFixed(0)).toString();
+    }
+    if (abs >= 10) {
+      return Number(number.toFixed(1)).toString();
     }
+    if (abs >= 0.01) {
+      return Number(number.toFixed(2)).toString();
+    }
+    if (abs === 0) {
+      return '0';
+    }
+
+    return number.toPrecision(2);
+  }
 
-    if (Math.abs(number) < 10) {
-      return number.toFixed(2);
+  /**
+   * Returns a formatted number string with a suffix.
+   *
+   * @param value {number|string}
+   * @return {string}
+   */
+  formatSuffixed(value) {
+
+    const abs = Math.abs(value);
+    value     = Number(value);
+
+    if (abs < 1e-9) {
+      return this.formatNumber(value * 1e9) + "μ";
+    }
+    if (abs < 1e-6) {
+      return this.formatNumber(value * 1e6) + "n";
+    }
```

Comment:
If I am not mistaken nano and micro are switched here:

```suggestion
    if (abs < 1e-9) {
      return this.formatNumber(value * 1e9) + "n";
    }
    if (abs < 1e-6) {
      return this.formatNumber(value * 1e6) + "μ";
    }
```